_posts/2024-01-12-kasten-openshift-monitoring.markdown

```diff
@@ -0,0 +1,815 @@
+            "3"
+          ],
+          [
+            1705335749.842,
+            "4"
+          ],
+          [
+            1705335779.842,
+            "4"
+          ],
+          ....
+```
+
+What we get here is a [range vector rather than an instant vector](https://satyanash.net/software/2021/01/04/understanding-prometheus-range-vectors.html). Which means exactly the same result before but evaluated at different times within the 10 last minutes. 
+
+The interesting thing that you may notice is the value change at the end of the range vector, we moved from 2 to 3 then 4. 
+
+20 minutes after the failed backup if you reevaluate this expression you will get only `4` for all the timestamps.
+
+Some prometheus function like `increase` work with range vector and in this case it squashes the result into an instant vector back.   
+
+for instance 
+
+```
+"query=increase(catalog_actions_count{policy='mcourcy-pacman-backup',status='failed'}[10m])"
+```
+
+Will return an instant vector showing the evolution of the number of failed backup within the 10 minutes :
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "mcourcy-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "mcourcy-pacman-backup",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "catalog-svc",
+          "status": "failed",
+          "type": "backup"
+        },
+        "value": [
+          1705356529.062,
+          "1.0526315789473684"
+        ]
+      }
+    ]
+  }
+}   
+```
+
+The result will be zero if you've passed the 10 minutes after the failure 
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "mcourcy-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "mcourcy-pacman-backup",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "catalog-svc",
+          "status": "failed",
+          "type": "backup"
+        },
+        "value": [
+          1705357729.062,
+          "0"
+        ]
+      }
+    ]
+  }
+}
+```
+
+The non integer value `1.0526315789473684` is because even if we see a change from 2 to 4 on a time range this is seen as a gradual change on the 10 minutes. If instead of `[10]` you choose `[5m]` the value would be higher because the change is seen as quicker hence more brutal.
```

Comment:
```suggestion
The non integer value `1.0526315789473684` is because even if we see a change from 2 to 4 on a time range this is seen as a gradual change on the 10 minutes. If instead of `[10m]` you choose `[5m]` in the query, the value would be higher because the change is seen as quicker hence more abrupt.
```